### DISCUSSION TITLE
How do I extract the watermark from the ts file?

### DISCUSSION THREAD

**Discussion by RayZan-J** [swesterfeld/audiowmark#35] [Q&A]
when i run this:
```bash
audiowmark hls-add vs0prep/out1.ts send1.ts 0123456789abcdef0011223344556677
```
How do I know it works correctly?
How do I extract the watermark from send1.ts ?

And , What should I do to get the watermark added every second?
Please help me.

**Comment by swesterfeld** [comment]
I've transferred this to discussions, as this is more a question than an actual problem with the code.

> How do I know it works correctly?

Typically you will have a complete system with a web server that is adding the watermark while it is sending the data and a client that plays the audio and/or video stream. One way to test that the system as a whole works correctly is to capture a few minutes of audio from your web browser and use `audiowmark get` on the audio file to see if a watermark can be found.

> How do I extract the watermark from send1.ts ?

Since `audiowmark` requires a wav (or flac/ogg/mp3) as input, you'll have to convert your .ts to that format. Usually you'll have more than one .ts file, so you can use the .m3u8 playlist that you use for your HLS streaming. Something like

```
ffmpeg -y -i vs0send/out.m3u8 vs0send.wav
```
where you place all the `send1.ts ... sendN.ts` in the .m3u8. Then `audiowmark get vs0send.wav` should return a watermark.

> What should I do to get the watermark added every second?

The watermark will be repeated periodically all ~50 seconds, however extraction from a shorter clip can work quite well, so this does not mean that you actually need 50 seconds of audio. Currently it is not possible to make the watermark itself shorter (except for using "short codes" maybe), but increasing the strength will make it more likely that a short clip from the watermark will be detected successfully. So for instance if `--strength 15` or `--strength 20` is acceptable from a quality perspective, then you can detect the watermark successfully even for shorter clips.

However, asking for the watermark to be detected reliably on 1 second of audio material would result in a strength requirement that is almost certainly to be unacceptable from a quality point of view.
